### DISCUSSION TITLE
是否可省略左侧插槽

### DISCUSSION THREAD

**Discussion by chaoyue1217** [opensumi/core#3502] [Q&A] [answered]
左侧的插槽部分是否可省去不要，当点击左侧的菜单时，直接根据菜单的不同替换中间内容区域的内容
![image](https://github.com/opensumi/core/assets/18662948/b524f095-1be5-4287-a66e-c67a4f31b7af)

**Comment by erha19** [comment, marked as answer]
@chaoyue1217 左侧面板你可以通过限定 Layout 的宽度来限制其展示，但是逻辑控制上需要做的自定义操作就比较多了，比较建议直接舍弃左侧面板，基于中间区域去创建自定义视图。

以 ide-startup 示例为例，上面这种方式可以参考修改这里的代码：

https://github.com/opensumi/ide-startup/blob/main/src/browser/render-app.ts#L30C26-L30C50
